exchanges/stream/websocket.go

```diff
@@ -683,6 +679,31 @@ func (w *Websocket) FlushChannels() error {
 	return nil
 }
 
+// updateChannelSubscriptions subscribes or unsubscribes from channels and checks that the correct number of channels
+// have been subscribed to or unsubscribed from.
+func (w *Websocket) updateChannelSubscriptions(c Connection, store *subscription.Store, incoming subscription.List) error {
+	subs, unsubs := w.GetChannelDifference(c, incoming)
+	if len(unsubs) != 0 {
+		prevState := store.Len()
+		if err := w.UnsubscribeChannels(c, unsubs); err != nil {
+			return err
+		}
+		if diff := prevState - store.Len(); diff != len(unsubs) {
```

Comment:
This doesn't feel right ...
1) Shouldn't Unsubscribe be erroring if it wasn't successful ?
2) Shouldn't the state of all subs in unsubs be changing ?
That said, it's not too bad either.

Reply:
> Shouldn't Unsubscribe be erroring if it wasn't successful ?

I think it should, I just added it as a catch all in the event we forgot to remove it from the store when it was successful. Then it should complain. _should_ 😬 

> Shouldn't the state of all subs in unsubs be changing

Now you are making me think this is all completely wrong 😆. This specifically didn't catch any issues. Can you suggest a better way as a back up check? Cause I am drooling at my screen trying to figure it out 🤤.

Reply:
Sorry for late reply on this.
I think I'd want see that ranging the subs and unsubs have changed State, and that store contains (or doesn't) each one. `ContainsAll` and `ContainsNone` or something.
I really don't like checking len as a catchall, because it could false positive.
We're not locking store when we do any of this, afterall.